src/main/blackbox/blackbox_io.c

```diff
@@ -145,12 +145,24 @@ void blackboxWriteUnsignedVB(uint32_t value)
 }
 
 /**
+ * ZigZag encoding maps all values of a signed integer into those of an unsigned integer in such
+ * a way that numbers of small absolute value correspond to small integers in the result.
+ *
+ * (Compared to just casting a signed to an unsigned which creates huge resulting numbers for
+ * small negative integers).
+ */
+static uint32_t zigzagEncode(int32_t value)
+{
+    return (uint32_t)((value << 1) ^ (value >> 31));
+}
+
```

Comment:
Sure, the decoder does that of course. But then you need a test for the
decoder as well :). I'll just add a bunch of input/expectation pairs as a
unit test.